src/search.rs

```diff
@@ -50,7 +50,13 @@ struct SearchTemplate {
 pub async fn find(req: Request<Body>) -> Result<Response<Body>, String> {
 	let nsfw_results = if setting(&req, "show_nsfw") == "on" { "&include_over_18=on" } else { "" };
 	let path = format!("{}.json?{}{}&raw_json=1", req.uri().path(), req.uri().query().unwrap_or_default(), nsfw_results);
-	let query = param(&path, "q").unwrap_or_default();
+	let mut query = param(&path, "q").unwrap_or_default();
+
+	if query.starts_with("https://www.reddit.com/") {
```

Comment:
This is a good start, but we need to support more than just `www.reddit.com` and `old.reddit.com`. Besides supporting `reddit.com` (no subdomain) we would need to support all of the other subdomains. `np.reddit.com` is an obvious one that comes to mind. I couldn't find a formal list of subdomains, so I think we should instead aim to determine of the query begins with `*.reddit.com` and strip the scheme and domain from that.

I think we should also support the `http` scheme as well if someone wants to put in a link URL that was generated before Reddit employed universal TLS.

In regex, what we'd look for is this prefix: `^https?://([^\./]+\.)*reddit.com/`

Reply:
If you want to avoid using `regex::Regex` unless absolutely necessary, you could write code that checks for the presence of the scheme, and if it is detected, then filter the string prefix:

```rust
if query.starts_with("https://") || if query.starts_with("http://")
```

Note the check on `https` comes first since nowadays `https` is a much more common scheme.

Reply:
I went with a regex (see da6b716e7587987349e0643cb932cf8f5d76fb19) but if manual parsing is preferred just say so and I will change it.